### ISSUE TITLE
PAM: Fix @{HOME} on Debian

### ISSUE THREAD

**Issue by ayekat** [ayekat/dotfiles#8]
For some reason, Debian does not set the the PAM_ITEM `@{HOME}` (or for some reason, Arch *does* set `@{HOME}`), which breaks the setup on Debian.

I need to investivate which component in the login procedure (I suspend *systemd-logind*) is responsible for setting that variable, and how exactly *logind*, PAM, and `pam_env.so` interact with each other.

**Comment by ayekat** [comment]
OK, it turns out that `@{HOME}` is not really an environment variable, but rather a special placeholder for the user's home as seen by PAM (according to the `pam_env.conf` man page). It is possible that this feature has been added only recently, though, which would explain why it's "still" missing in Debian.

**Comment by ayekat** [comment]
OK, it turns out that Debian is simply running an ancient version of PAM (1.1.8), whereas the [commit](https://git.fedorahosted.org/cgit/linux-pam.git/commit/?id=73bdfac8c091492f466342feb8f2f5daa2f4c39b) that introduced `@{HOME}` and `@{SHELL}` was made for PAM 1.2 (two years ago).

Unfortunately, it seems that the Debian package for PAM has been [abandoned](http://metadata.ftp-master.debian.org/changelogs/main/p/pam/pam_1.1.8-3.5_changelog), and Testing and Experimental both [only provide 1.1.8](https://tracker.debian.org/pkg/pam), and given that the freeze for Stretch is only half a month away, we quite likely won't receive any PAM update in a reasonable future.

The only workaround seems to create a Debian-specific `~/.pam_environment` that uses a butt-ugly work-around (defining `$HOME` manually) &mdash; or simply consider Debian as broken and thus unsupported for these dotfiles.

**Comment by ayekat** [comment]
As of 83ea418, Debian support has been dropped, so there is no point in digging further on this.

**Comment by ayekat** [comment]
Reopening this in an effort to add back Debian support.

**Comment by ayekat** [comment]
[Florian Vessaz appears to have created a Debian package for PAM 1.3](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=821408#10), so it might be worth checking that one out.

**Comment by ayekat** [comment]
There is now a [debian](https://github.com/ayekat/dotfiles/tree/debian) branch for this case. It's ugly, but waiting >2 years for things to be fixed is a bit extreme.

**Comment by ayekat** [comment]
Using the `DEFAULT` and `OVERRIDE` directives in `.pam_environment` (see commit 6dec7a9f26), we can now use a single file for the ugly workaround. The debian branch is thus no longer needed (deleting).

**Comment by ayekat** [comment]
Reading that ML discussion again, it appears that the original maintainer has finally decided to start maintaining the PAM package again. Unfortunately, this was only after 10 (Buster) was released, so we won't see any updates on this until 11 (Bullseye).

**Comment by ayekat** [comment]
See #32. Closing this issue as "no longer relevant".
